openmc_plotter/main_window.py

```diff
@@ -1209,20 +1151,16 @@ def closeEvent(self, event):
         self.saveSettings()
 
     def saveSettings(self):
+        if self.model.statepoint:
+                self.model.statepoint.close()
 
-        if len(self.model.previousViews) > 10:
-            self.model.previousViews = self.model.previousViews[-10:]
-        if len(self.model.subsequentViews) > 10:
-            self.model.subsequentViews = self.model.subsequentViews[-10:]
-
-        # get hashes for geometry.xml and material.xml at close
-        self.model.mat_xml_hash = hash_file('materials.xml')
-        self.model.geom_xml_hash = hash_file('geometry.xml')
-
+        pickle_data = {
+            'version': self.model.version,
+            'currentView_ind': self.model.currentView.view_ind,
+            'statepoint': self.model.statepoint
+        }
```

Comment:
The goal of this was to prevent having to store the large arrays with the cell/mat/property data to the pickle file smaller. Plus the goal of this PR was to separate the model-dependent properties from model-independent properties. Though I can see how this property is in the gray zone. If the material/cells IDs change, then the mapping of colors to materials/is no longer valid. I will play with it and see what can be done. Might have to go back to using hashes in that if the hashes are unchanged, the entire current view can be reused, but if the hashes differ, then use only the independent view settings.

Reply:
Ah, and I am just remembering that we stored the full `model` in the past which is where that large amount of excess data came from. So storing the full currentView should be no issue.